nano/node/election.cpp

```diff
@@ -388,24 +388,26 @@ nano::election_vote_result nano::election::vote (nano::account const & rep, uint
 	else
 	{
 		auto last_vote_l (last_vote_it->second);
```

Comment:
This does not look right to me, the two new if statements, do not do exactly the same thing as the one old if statement.

There re 2 problems:
1) a missing equal sign in `last_vote_l.timestamp > timestamp_a`
2) the logic is different even ignoring the missing equal sign

For 2) We had:
```
if (a && b)
    return;
```

And we replaced it with:
```
if (a)
    return;
if (b)
    return;
```

But it should instead be:
```
if (a) {
  if (b) {
    return;
  }
}

Reply:
In general you would be right, but in this case timestamps are compared in both `a` and `b`.
So there are 3 possibilities:

1. `last_vote_l.timestamp < timestamp_a` - always process further
2. `last_vote_l.timestamp > timestamp_a` - always return, because `b` is false - `!(last_vote_l.timestamp == timestamp_a)`. Equal sign is omitted on purpose here.
3. `last_vote_l.timestamp == timestamp_a` - depending on hash, that's what `if (b)` checks.

Reply:
Nice explanation. I agree.
 One final improvement would be to to convert '!(last_vote_l.hash < block_hash_a)' to 'last_vote_l.hash >= block_hash_a'.

Reply:
Yeah, but it seems block_hash has only operator< defined - https://github.com/nanocurrency/nano-node/blob/a0dd56cc8ad07832d4f494ab627dae6eb1472089/nano/lib/numbers.hpp#L105.

Reply:
OK, no big deal then.